nvm/pmemobj/tuple.py

```diff
@@ -0,0 +1,96 @@
+import collections
+import sys
+
+from .compat import recursive_repr, abc
+from .list import PersistentList
+from _pmem import ffi
+
+TUPLE_POBJPTR_ARRAY_TYPE_NUM = 50
```

Comment:
This approach is used in dict, list, pool and now tuple. If there's a solution to this (Python doesn't have `typeid`, but you can still do `hash(type(var))` and get a hash based on the type of the variable `var`), it's going to be implemented in a separate pull request (since it's a widespread pattern).